scip/api/population.py

```diff
@@ -0,0 +1,80 @@
+from salmon_occurrence import ConservationUnit, Taxon, Reference, Population, Phenology
+from sqlalchemy_sqlschema import maintain_schema
+from sqlalchemy_sqlschema.sql import get_schema
+from sqlalchemy import func
+
+
+def population(
+    session,
+    overlap=None,
+    common_name=None,
+    scientific_name=None,
+    subgroup=None,
+    name=None,
+):
+    """Return information about salmon populations in the database that fulfill
+    specified parameters.
+
+    :param session: (sqlalchemy.orm.session.Session) a database Session object
+    :param overlap: a WKT string specifying a geometry that overlaps with the desired populations
+    :param common_name: common name of the salmon species of interest
+    :param scientific_name: scientific name of the salmon species of interest
+    :param subgroup: parameter designating a sub-species taxon, such as `lake` or `river` for sockeye salmon
+    :param name: name of the conservation unit that encloses the population's range
+
+    :return: a list of objects representing salmon populations that fulfill the given parameters. In
+        addition to `common_name`, `scientific_name`, `subgroup`, and `name`, two geoJSON strings describing
+        the geometry of the population7s extent are provided. `boundary` describes the outline of the extent,
+        and `outlet` provides the most downstream point of the extent according to the RVIC routed flow data.
+
+    """
+    # TODO: verify input arguments
```

Comment:
Yes, instead of the `call` method. I bet we could come up with something pretty tight and simple to use for this kind of checking generally.

Reply:
For the Flask rework PR